numpy/core/function_base.py

```diff
@@ -1,11 +1,25 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
 
+def index_deprecate(i):
+    try:
+        i = operator.index(i)
+    except TypeError:
+        i = int(i)
+        msg = "Indexing with a float is deprecated "\
+        "and will be removed in the future."
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
```

Comment:
If you do this, make it a default of `2` and then add 1 to it before passing to `warn` -- the idea being that the `stacklevel` users type should be relative to them, not relative to inside this function.